typescript/src/services/TsParser/getDecorators.ts

```diff
@@ -6,14 +6,11 @@ export function getDecorators(declaration: Declaration) {
   if (declaration.decorators) {
     return declaration.decorators.map(decorator => {
       const callExpression = getCallExpression(decorator);
-      if (!callExpression) {
-        throw new Error(`Expected decorator to be a call expression: ${decorator.getText()}`);
-      }
       return {
-        argumentInfo: callExpression.arguments.map(argument => parseArgument(argument)),
-        arguments: callExpression.arguments.map(argument => argument.getText()),
+        argumentInfo: callExpression && callExpression.arguments.map(argument => parseArgument(argument)),
+        arguments: callExpression && callExpression.arguments.map(argument => argument.getText()),
         expression: decorator as Decorator,
-        name: callExpression.expression.getText(),
+        name: callExpression ? callExpression.expression.getText() : decorator.expression.getText(),
```

Comment:
I'll await your next PR but for what it is worth I think it would be simpler to use an `if`:

```
if (callExpression) {
  return ... // the original object
} else {
  return {
    name: decorator.expression.getText()
  };
}
```

And of course we will need tests :-)